src/viam/app/location/client.py

```diff
@@ -0,0 +1,711 @@
+            self.organization_owner = fragment.organization_owner
+            self.public = fragment.public
+            self.created_on = fragment.created_on.ToDatetime() if fragment.HasField("created_on") else None
+            self.organization_name = fragment.organization_name
+            self.robot_part_count = fragment.robot_part_count
+            self.organization_count = fragment.organization_count
+            self.only_used_by_owner = fragment.only_used_by_owner
+            return self
+
+        id: str
+        name: str
+        fragment: Optional[Mapping[str, Any]]
+        organization_owner: str
+        public: bool
+        created_on: Optional[datetime]
+        organization_name: str
+        robot_part_count: int
+        organization_count: int
+        only_used_by_owner: int
+
+    def __init__(self, channel: Channel, metadata: Mapping[str, str], location_id: Optional[str] = None):
+        """Create a `LocationClient` that maintains a connection to app.
+
+        Args:
+            channel (grpclib.client.Channel): connection to app.
+            metadata (Mapping[str, str]): Required authorization token to send requests to app.
+            location_id (Optional[str]): Default location ID.
+        """
+        self._metadata = metadata
+        self._location_client = AppServiceStub(channel)
+        self._location_id = location_id
+
+    _location_client: AppServiceStub
+    _metadata: Mapping[str, str]
+    _location_id: str
+
+    async def get_user_id_by_email(self):
+        raise NotImplementedError()
+
+    async def create_organization(self):
+        raise NotImplementedError()
+
+    async def list_organizations(self):
+        raise NotImplementedError()
+
+    async def list_organizations_by_user(self):
+        raise NotImplementedError()
+
+    async def get_organization(self):
+        raise NotImplementedError()
+
+    async def get_organization_namespace_availability(self):
+        raise NotImplementedError()
+
+    async def update_organization(self):
+        raise NotImplementedError()
+
+    async def delete_organization(self):
+        raise NotImplementedError()
+
+    async def list_organization_members(self):
+        raise NotImplementedError()
+
+    async def create_organization_invite(self):
+        raise NotImplementedError()
+
+    async def update_organization_invite_authorizations(self):
+        raise NotImplementedError()
+
+    async def delete_organization_member(self):
+        raise NotImplementedError()
+
+    async def delete_organization_invite(self):
+        raise NotImplementedError()
+
+    async def resend_organization_invite(self):
+        raise NotImplementedError()
+
+    async def create_location(self, organization_id: str, name: str, parent_location_id: Optional[str] = None) -> Location:
+        """Create and name a location under the specified organization and parent location.
```

Comment:
As of right now, I am able to successfully call this method, as long as the org_id is from the org that the initial location I used to auth belongs to it. Should I assume this won't be the case then? Also I though if should work if we're authed as an org owner?

Reply:
Ah interesting, I may have gotten myself confused then! Yeah, maybe we should just update `raises` to indicate that we raise if we pass an org_id that we don't have permissions in? 

I am curious how it will work if we try to create top level locations in the org after auth changes go in, since we'll no longer be an owner of the org. but we're figuring out solutions to that now so no need to dig into it.

Reply:
I'm also curious how this will play out when the auth changes take place. Not sure what to expect at this point.